src/modules/service-catalog/components/useItemFormFields.spec.ts

```diff
@@ -0,0 +1,394 @@
+import { act, renderHook } from "@testing-library/react-hooks";
+import type { ServiceCatalogItem } from "../data-types/ServiceCatalogItem";
+import { useItemFormFields } from "./useItemFormFields";
+
+describe("useItemFormFields", () => {
+  const mockedItem: ServiceCatalogItem = {
+    id: 1,
+    name: "Test Item",
+    description: "Test Description",
+    form_id: 1,
+  };
+
+  const textField = {
+    id: 1,
+    type: "text",
+    description: "Test Description",
+    title_in_portal: "Test Title",
+    editable_in_portal: true,
+    required_in_portal: true,
+  };
+
+  const lookupField = {
+    id: 2,
+    type: "lookup",
+    description: "Service",
+    title_in_portal: "Service",
+    editable_in_portal: true,
+    relationship_target_type: "standard::service_catalog_item",
+    required_in_portal: true,
+  };
+
+  const expectedTextField = {
+    id: 1,
+    type: "text",
+    name: "custom_fields_1",
+    description: "Test Description",
+    label: "Test Title",
+    options: undefined,
+    required: true,
+    relationship_target_type: undefined,
+    error: null,
+    value: null,
+  };
+
+  const expectedLookupField = {
+    description: "Service",
+    editable_in_portal: true,
+    id: 2,
+    relationship_target_type: "standard::service_catalog_item",
+    required_in_portal: true,
+    title_in_portal: "Service",
+    type: "lookup",
+  };
+
+  const additionalTextField = {
+    id: 3,
+    type: "text",
+    name: "custom_fields_3",
+    description: "Test Description 3",
+    label: "Test Title 3",
+    options: [],
+    required: true,
+    relationship_target_type: undefined,
+    error: null,
+    value: "Test Value 3",
+  };
+
+  const mockedBaseLocale = "en-us";
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("should initialize with default values", async () => {
+    const mockedFormResponse = {
+      ticket_form: {
+        id: 1,
+        ticket_field_ids: [1, 2],
+        active: true,
+      },
+    };
+
+    const mockedTicketFieldResponse = {
+      ticket_fields: [textField, lookupField],
+    };
+
+    global.fetch = jest.fn((url) => {
+      if (url.includes("/ticket_forms")) {
+        return Promise.resolve({
+          json: () => Promise.resolve(mockedFormResponse),
+          status: 200,
+          ok: true,
+        });
+      } else {
+        return Promise.resolve({
```

Comment:
Can we target the URLs more specifically so we don't rely on the current implementation:
```ts
return Promise.resolve({
  json: () => Promise.resolve("/api/v2/ticket_forms/1"
    ? formResponse
    : "/api/v2/ticket_fields?locale=en-us"
    ? ticketFieldResponse
    : {}),
  status: 200,
  ok: true,
});
```

Same in the following ones.